### DISCUSSION TITLE
macOS x86_64 emulation and vz virtualization

### DISCUSSION THREAD

**Discussion by miquecg** [abiosoft/colima#753] [Q&A] [answered]
Hi,

I have a running colima instance on macOS 13.4 ARM host, configured with `x86_64` cpu, `vz` virtualization, `rosetta` enabled and `virtiofs`. It's running fine but I'm wondering if this setup is even possible.

Reported status matches what is configured in the YAML file but I suspect it's actually running on `qemu`. I've noticed that in some debug messages when booting instance up. How can I double check this?

Thanks!

**Comment by sebastianrothe** [comment]
Why do you think that this is not possible?

VZ is the virt framework from Apple.
Rosetta is the emulation framework for x86 from Apple.

My guess is, that your cpuType is ignored, because you use `vz`. `cpuType` is only possible with qemu.

You can double-check with `ActivityManager`. You will either see a process `qemu` or `VirtualMachineService` and Kind `Apple`.

**Reply by miquecg** [reply]
As I was suspecting, `qemu` is actually running hence my assumption that this setup is not yet possible with macOS Virtualization Framework or some setting I'm using is not compatible.

> My guess is, that your cpuType is ignored, because you use vz. cpuType is only possible with qemu.

Sorry, I meant architecture is `x86_64`. `cpuType` setting is set to `host` as per defaults.

**Reply by 66Ton99** [reply]
colima status --profile x86
show:
```
INFO[0022] colima [profile=x86] is running using macOS Virtualization.Framework
INFO[0022] arch: x86_64
INFO[0022] runtime: docker
INFO[0022] mountType: virtiofs
INFO[0022] socket: unix:///Users/ton/.colima/x86/docker.sock
```
But `ActivityManager` shows `qemu` :)

**Comment by sebastianrothe** [comment, marked as answer]
Is there a reason you have to use `x86_64`? You should go native and the use the emulation framework `rosetta` same like Apple does.

**Reply by miquecg** [reply]
Not really that I know of yet. It was partly because a misunderstanding of how emulation is achieved. I didn't realize setting `arch` to `x86_64` would force to use qemu emulation instead of Rosetta, which I guess is a better option on macOS given it works (I've seen reports from people with problems related to Rosetta emulation). Gonna give a try anyway.

Thanks.

**Comment by buncis** [comment]
I think `--arch x86_64 ` forcing colima to use qemu instead vz, and therefore it can't use rosetta

**Reply by pPanda-beta** [reply]
But any code which is taking such crucial decision should show warnings. 
For me same happened, silently it started qemu.

I checked by 
```sh
ps aux  |  grep qemu

# Output: /opt/homebrew/bin/qemu-system-x86_64 ... -name lima-colima -pidfile /Users/****/.colima/_lima/colima/qemu.pid ... 
```
